dist/srpm/rhc.spec.in

```diff
@@ -122,12 +116,25 @@ ln -sf ../yggdrasil/config.toml %{buildroot}%{_sysconfdir}/rhc/config.toml
 %endif
 %endif
 
+%pre
+if [ $1 -eq 2 ]; then
+    if [ -f /etc/rhc/config.toml ]; then
+        cp /etc/rhc/config.toml /etc/yggdrasil/config.toml.migrated
+    fi
+fi
```

Comment:
Since there is nothing that removes `/etc/rhc/config.toml`, this snippet will run not only at the first time it is upgraded, but every time. The result is that the generated `/etc/yggdrasil/config.toml` from this is generated at each upgrade, even overwriting changes made by the users.

Reply:
Does the file get left in place when the old rhc is removed? The RPM marks the file as `%config`; doesn't that move the file to `/etc/rhc/config.toml.rpmsave`?

Reply:
I think I figured it out. I forced an EL 10 system to install rhc-0.2.4 from EL9. Then I upgraded to the rhc RPM from this PR. The file `/etc/rhc/config.toml` is gone after the upgrade, and the file `/etc/yggdrasil/config.toml` contains the modified contents of the rhc config file.

I noticed during testing that uninstalling `yggdrasil` leaves the file `/etc/yggdrasil/config.toml.rpmsave` because the file was modified after the package was installed.